### PR TITLE
add profile photo uploader to work/account

### PR DESCRIPTION
**Pull request by jerminatorhits**
## Description
Why did you write this code?
add profile photo uploading to account page

## How to Test
- [ ] Visit work/account
- [ ] upload photo successfully

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?
The old site straight up does not display this component on Mobile and Tablet
Right now, it is displayed for all tabs (it'll just kick down to the bottom of screen).
This might be kind of noisy for the user to always have it there. Might consider making another tab separately for it or wait on designs for other alternatives.

## Learnings
Did you learn anything here you want to share with the team?

### CODE REVIEW

**Comment by kcvan** [comment]
have re figured out a way to upload it without having a refresh?

File: src/pages/Account/Account.tsx [hunk 1]
```diff
@@ -82,6 +82,11 @@ const Account = () => {
                     <Route component={NotFound} />
                   </Switch>
                 </Col>
+                <Col md={1} />
```
Comment (kcvan) [inline src/pages/Account/Account.tsx]: if it's only one parameter you can do `md="1"`, i think if you just do `{1}` it expects { size: '', etc: 'etc' }

Comment (jerminatorhits) [reply src/pages/Account/Account.tsx]: ahh gotcha.

also, I don't actually know when to use md, sm, lg, and if when it is good to be specific. I want to declare the column size based on 12 and so I stick to `md` for that declaration. ¯\_(ツ)_/¯

File: src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import styled from 'styled-components';
+import { color, typography } from 'styled/utils';
+
+const ProfilePhotoUploaderContainer = styled.div`
+  .photo-uploader-container {
```
Comment (tc) [inline src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: i assume this is a copy and paste from old component?

Comment (jerminatorhits) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: mhm

Comment (tc) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: it's ok it is a copy/pasta but if it's completely unchanged, can't we just use the existing component?

Comment (jerminatorhits) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: ooo the old one has media queries where there's a display none when the screen is less than desktop. In this implementation in an effort to shy away from media queries, the photo uploader is always shown and I removed that display none property.

If the old component was used, there would be an empty `<Card />` component (it wraps ProfilePhotoUploader) on Tablet and Mobile.

Comment (jerminatorhits) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: image

Comment (tc) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: ok, ideally we want to stick to the components front already provided for us to avoid re-writing CSS.
see:
screen shot 2019-02-27 at 3 22 51 pm

The photo uploader can just be a button "Change Photo" vs drag and drop to save effort.

**Comment by jerminatorhits** [comment]
@kcvan The photo updates for me without having to refresh

File: src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts (hunk 2 above)
Comment (jerminatorhits) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: Currently, the networking to upload to our backend and S3 is happening in the `<PhotoUploader />` component. Should I try to route an input field through that component or perhaps make a new gql endpoint just for photo uploading?

Comment (tc) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: can we change the styling without changing the functionality?

Comment (kcvan) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: i think we can do the same thing we're doing in the old header where it checks the db, firebase, and then returns the default photo.  that way, we can ensure we get an updated photo once the DB updates

File: src/pages/Account/Account.tsx (hunk 1 above)
Comment (kcvan) [reply src/pages/Account/Account.tsx]: its the same as media queries, you can think of sm = mobile, md = tablet, lg = desktop, xl = very large.  whatever you specify as the smallest will propagate up to largest screen.  usually if you're on mobile, if you're inside a row it normally has `flex-wrap` if i remember correctly youll have to do something like `sm="12"` if you want the entire column to span the width of the screen, or else it'll stack and then wrap.

File: src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts (hunk 2 above)
Comment (jerminatorhits) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: don't think so since the styles are coupled within the PhotoUploader Component, I can try to recreate new component: the Dropzone component that use bootstrap styles only?

Comment (tc) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: let's just hold off on it for now. this PR looks fine with the current profile upload design.

Comment (jerminatorhits) [reply src/components/work/ProfilePhotoUploader/ProfilePhotoUploader.container.ts]: ok sg
